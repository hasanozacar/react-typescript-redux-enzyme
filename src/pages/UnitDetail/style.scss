.container {
  display: flex;
  flex-direction: column;
}

.title {
  align-self: center;
}
.image {
  max-width: 600px;
  max-height: 200px;
  padding: 20px;
}
.logo-container {
  width: 100%;
  height: 50%;
  align-self: center;
  margin-top: 30px;
}
