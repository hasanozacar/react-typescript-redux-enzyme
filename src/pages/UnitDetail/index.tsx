import React, { Component } from 'react';
import './style.scss';
import { connect } from 'react-redux';
import { RootObject } from '../../types/dataTypes';
import { Descriptions } from 'antd';
import logo from '../../global/assets/sultan.jpeg'

export type UnitDetailProps = {
  unitDetail?: RootObject | undefined,
};

class UnitDetail extends Component<UnitDetailProps> {
  constructor(props: UnitDetailProps) {
    super(props);

    this.state = {
    };
  }

  renderCostItems = (key: string, value: any) => (
    <div>
      <h3>
        {`${key}: ${value || 'Unknown'}`}
      </h3>
    </div>
  )

  renderElements = (data: RootObject | undefined) => {
    if (typeof data === "object") {
      const keys = Object.keys(data);
      return (
        keys.map(k => (k !== 'cost' ? (
          <Descriptions.Item label={k.toUpperCase()}>
            <h3>
              {data[k]}
            </h3>
          </Descriptions.Item>
        ) :
          <Descriptions.Item label="Cost">
            {this.renderCostItems('Wood Cost', data.cost?.Wood)}
            {this.renderCostItems('Food Cost', data.cost?.Food)}
            {this.renderCostItems('Gold Cost', data.cost?.Gold)}
          </Descriptions.Item>)));
    }
  }

  render() {
    return (
      <div className="container">
        <h2 className="title"> Unit Detail Page</h2>
        <img src={logo} className="image" alt="Logo" />
        <Descriptions
          bordered
          column={{ xxl: 1, xl: 1, lg: 1, md: 1, sm: 1, xs: 1 }}
        >
          {this.renderElements(this.props?.unitDetail)}
        </Descriptions>
      </div>
    );
  }
}

const mapStateToProps = (state: any) => ({
  unitDetail: state.units?.unitDetail
});


export default connect(mapStateToProps, null)(UnitDetail);
