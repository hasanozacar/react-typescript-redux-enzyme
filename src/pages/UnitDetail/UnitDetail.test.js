import React from "react";
import UnitDetail from ".";
import { Provider } from "react-redux";
import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import "../../setupTests";
import * as Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe("UnitDetail", () => {
  const mockStore = configureStore();
  const store = mockStore({
    news: {
      title: "Mock Title",
      urlToImage: "www.google.com",
      description: "Mock Description",
      url: "bbc.com",
    },
    loading: false,
  });
  const wrapper = mount(
    <Provider store={store}>
      <UnitDetail
        unitDetail={{
          id: 104,
          name: "Elite Woad Raider",
          description: "Upgraded Woad Raider",
          expansion: "Age of Kings",
          age: "Imperial",
          cost: {
            Gold: 25,
          },
          build_time: 10,
          reload_time: 2,
          movement_rate: 1.38,
          line_of_sight: 5,
          hit_points: 80,
          attack: 13,
          armor: "0/1",
          attack_bonus: ["+3 eagles", "+3 buildings"],
        }}
      />
    </Provider>
  );
  it("renders without crashing", () => {
    expect(wrapper.find(UnitDetail).length).toEqual(1);
  });
});
