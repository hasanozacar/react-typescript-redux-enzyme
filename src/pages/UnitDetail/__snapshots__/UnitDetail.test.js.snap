// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly <UnitDetail> 1`] = `ReactWrapper {}`;
